services/provider/server/storageclaim.go

```diff
@@ -79,7 +71,12 @@ func (s *storageClassClaimManager) Create(ctx context.Context, consumer *ocsv1al
 		},
 	}
 
-	err := s.client.Create(ctx, storageClassClaimObj)
+	err := controllerutil.SetOwnerReference(consumer, storageClassClaimObj, s.client.Scheme())
```

Comment:
This will not set the `BlockOwnerDeletion` flag. only SetControllerReference will but in that case we have an issue of a wrong controller set on the owner, Please revert

Reply:
The manual setting of ownRef was not working. so we added this code when we found issue during Create operation. i brought back the functionality of BlockOwnerDeletion. as this is already testing code can we stick to this one, if we need to go back to the previous version need to check what is going wrong in setting ownRef and need more testing.

Reply:
As discussed, changes are done PTAL